pydantic_ai_slim/pydantic_ai/models/anthropic.py

```diff
@@ -328,10 +333,18 @@ async def _messages_create(
         try:
             extra_headers = model_settings.get('extra_headers', {})
             extra_headers.setdefault('User-Agent', get_user_agent())
-            if beta_features:
-                if 'anthropic-beta' in extra_headers:
-                    beta_features.insert(0, extra_headers['anthropic-beta'])
-                extra_headers['anthropic-beta'] = ','.join(beta_features)
+            if beta_features or structured_output_beta_required:
+                new_features = list(beta_features)
+                if structured_output_beta_required:
+                    new_features.append(_STRUCTURED_OUTPUTS_BETA)
+                extra_headers['anthropic-beta'] = self._format_beta_header(
+                    extra_headers.get('anthropic-beta'),
+                    new_features,
+                )
+
+            extra_body = cast(dict[str, Any] | None, model_settings.get('extra_body'))
+            if output_format is not None:
+                extra_body = self._merge_output_format_extra_body(extra_body, output_format)
```

Comment:
We should bump the dependency so we can pass this directly instead of via extra body